### PR TITLE
signal.lti: Remove deprecated cross-system properties

### PR DESCRIPTION
**Pull request by befelix**
Cross-system properties/setters have been deprecated in version [0.18](https://github.com/scipy/scipy/blob/master/doc/release/0.18.0-notes.rst#deprecated-features) with commit 6939e33b11dda6815fd851b9ca943625ab606971. This commit follows up with removing the deprecated code for the 1.0 release.

### CODE REVIEW

File: doc/release/1.0.0-notes.rst [hunk 1]
```diff
@@ -75,6 +75,14 @@ has been changed to True.
 
 `scipy.special.gammaln` does not accept complex arguments anymore.
 
+The cross-class properties of `scipy.signal.lti` systems have been removed.
+The following properties/setters have been removed:
+
+Name - (accessing/setting has been removed) - (setting has been removed)
+* StateSpace - (`num`, `den`, `gain`) - (`zeros`, `poles`)
```
Comment (rgommers) [inline doc/release/1.0.0-notes.rst]: for this to render correctly, need a blank line above the list and double backticks around the names

**Comment by rgommers** [comment]
Thanks @befelix, looks good to me. Will leave this open for a little to give others time to comment.

**Review by larsoner — APPROVED** [review]
LGTM +1 for merge

**Comment by Eric89GXL** [comment]
@e-q do you have time to look?

**Comment by e-q** [comment]
Sorry, I've been buried under thesis writing for a while. From a quick glance this all looks straightforward and good.

**Comment by rgommers** [comment]
Okay, in it goes. Thanks @befelix, all.

**Comment by rgommers** [comment]
> Sorry, I've been buried under thesis writing for a while. 

Good luck with that Eric!

**Pull request merged**